### DISCUSSION TITLE
Major rewrite

### DISCUSSION THREAD

**Discussion by TimGels** [fossmium/OneDrive-Cloud-Player#79] [General]
We have been discussing a rewrite and overhaul of the application for a while now. A downside to such a big rewrite is that it is very costly in terms of our time. Because we work on this in our limited spare time, and we do not generate any income from this as a form of compensation, it is not something that we can or should take lightly. However, lately it has come to my attention that the application is suffering from a lot of crashes and other instability issues. In fact, a lot more than I am comfortable with. People should be able to use this application without constant crashes and that is currently not what I am observing, given the current failure rate.

It is not surprising to me that the application is suffering from these stability issues as it was originally developed as a school assignment for C#. The application is in need of underlying architectural changes to improve the stability and make it more maintainable. We can also take this rewrite to overhaul the UI to make it more usable for Xbox users as well.

Due to the reasons stated above, I want to propose fast-tracking a rewrite of the application with the goal to make it more stable and maintainable.

**Comment by JohannesKauffmann** [comment]
If the application is crashing a lot, I think we should fix those crashes first before rewriting everything. People having crashes right now won't be happy to hear those crash will eventually be fixed in a year... hopefully. So I think we should focus on that first.

I'm not against rewriting parts of the application, but IMO this should be an incremental process.

**Reply by TimGels** [reply]
Then I would like to suggest that we start with implementing some sort of crash analytics (preferred to be something users can opt out of) so we can monitor the crashing a bit better. After that, we can start with making the login/ authentication part of the application more stable. Once we have done that, we can decide what part will be next. What do you think?

**Reply by JohannesKauffmann** [reply]
Yes, crash analytics is needed. Another issue we should fix quickly is the app crashing when trying to log in with an account that has no OneDrive associated with it. Afterwards we can look into more structural changes.
